runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,357 @@
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = ClientDestinationInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    DestinationInfoVal clientDestinations;
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                        if (clientDestinations.getDestinationIdsList().contains(destination)) {
+                            throw new IllegalArgumentException("Destination already exists");
+                        }
+
+                        clientDestinations = clientDestinations.toBuilder().addDestinationIds(destination).build();
+                    } else {
+                        clientDestinations = DestinationInfoVal.newBuilder().addDestinationIds(destination).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+                } catch (TransactionAbortedException tae) {
+                    log.error("Unable to add destination, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Unable to add destination");
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Add a list destination to LogReplicationSourceMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param remoteDestinations Clusters which the logicalGroup should be replicated to
+     */
+    public void addDestination(String logicalGroup, List<String> remoteDestinations) {
+        Preconditions.checkArgument(isValid(logicalGroup), "logicalGroup is null or empty");
+        Preconditions.checkArgument(isValid(remoteDestinations), "remoteDestinations is null or empty");
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = ClientDestinationInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    DestinationInfoVal clientDestinations;
+
+                    boolean hasNullOrEmptyOrDuplicates =
```

Comment:
null and empty have already been checked earlier, do we need to check them again?

Also, we should not fail the operation if there are duplicate destinations IMO.. instead de-duplicate them.  Let me know if you think otherwise.